python/cudf/cudf/core/series.py

```diff
@@ -1518,6 +1581,41 @@ def _concat(cls, objs, axis=0, index=True):
             [name] = names
         else:
             name = None
+
+        if len(objs) > 1:
+            dtype_mismatch = False
+            for obj in objs[1:]:
+                if (
+                    obj.null_count == len(obj)
+                    or len(obj) == 0
+                    or is_categorical_dtype(obj.dtype)
+                    or is_categorical_dtype(objs[0].dtype)
```

Comment:
Can we similarly check `obj._column` and `objs[0]._column` for `CategoricalColumn` instead of the more expensive dtype check here?